src/karapace/avro_dataclasses/introspect.py

```diff
@@ -128,7 +135,7 @@ def _field_type(field: Field, type_: object) -> AvroType:  # pylint: disable=too
 T = TypeVar("T", str, int, bool, Enum, None)
 
 
-def transform_default(type_: type[T], default: T) -> str | int | bool | None:
+def transform_default(type_: type[T] | str, default: T) -> str | int | bool | None:
```

Comment:
This is pure typing correction. It looks that primitive types use string instead of `type[T]`.

Reply:
Tracking this to here:

https://github.com/python/typeshed/blob/6feca188689e14cdacd8dbf8bc9cf0635993027a/stdlib/dataclasses.pyi#L111

When originally authored, this type was `type`, only changed recently and I guess propagated in most recent mypy bump:

https://github.com/python/typeshed/commit/6b2033a783ad4f963979c761f32b14d1d9374bc5

Reply:
`_field_type` doesn't give a type error because it accepts `object` and makes no assumptions. It's probably somewhat likely to break when the new deferred annotations arrive, but I wouldn't deal with that until it's possible to test with.